notebooks/stitches_dev/nearest_neighbor_matching.R

```diff
@@ -95,9 +113,52 @@ out <- cbind(target_data, matched)
 return(out)
 
   
+}
+###############################################################################
+# A helper function to remove false duplicate matches in the historical period. For
+# example, target 1850 gets 1872 data from realization 13 of SSP126 and SSP585.
+# The metadata of these archive values are different, but the actual data 
+# values are identical because we just pasted in the same historical data to 
+# every Experiment. So this function keeps only the first match.
+# 
+# Args: matched_data: data output from match_neighborhood.
+#
+# Returns: a data frame of matched data with the same structure as the 
+# input, with false duplicates in the historical period dropped.
+drop_hist_false_duplicates <- function(matched_data){
+  matched_data %>%
+    # Because smoothing with window =9 has occured, 
+    # historical is actually 2010 or earlier: the chunks
+    # that had purely historical data in them and none 
```

Comment:
should we make this some sort of window that can change based on the chunk size? or tag that development for later, when we have the other pieces figured out?

Reply:
MIght as well do it now so that the cutoff year is `2015-[ (window_size_from_matched_data-1)/2]`